### PR TITLE
Perf doc

### PR DESCRIPTION
**Pull request by graydon**
This just adds a new (preliminary!) performance.md doc as we've been discussing, that talks explicitly about sources of load on stellar-core nodes and ways to configure it to control for them.

Additions welcome. I figured I'd get a sketch done and we could build it out with contributions from operators.

(There are 2 doc-maintenance commits in here too, the only one about performance.md is the third commit)

### CODE REVIEW

File: docs/software/performance.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+## Purpose of this document
+
+This document describes performance considerations for administrators and developers of `stellar-core`. It may also be interesting for a broader audience interested in understanding system performance and scaling limits and avenues for future work.
+
+## Overview
+
+The performance of a `stellar-core` node varies in two main dimensions:
+
+  1. _What it is configured to do_:
+    * As discussed in the [admin.md](./admin.md) file, nodes may be configured as watchers, archivers, basic validators or full validators. These roles have different performance costs.
```
Comment (tomquisel) [inline docs/software/performance.md]: These list items aren't rendering right (I think you just need a newline)

File: docs/software/performance.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    * Each such role may have a variety of options enabled or disabled at varying costs to the node.
+
+  2. _How it is physically and logically configured to do its job_:
+    * The physical hardware -- disks, CPUs, memory and networking equipment -- supporting the node can substantially affect performance.
```
Comment (tomquisel) [inline docs/software/performance.md]: ditto

File: docs/software/performance.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+  1. The `BUCKET_DIR_PATH` config option sets the location that `stellar-core` places its buckets while (re)writing the bucket list. This should be located on a relatively fast, low-latency local disk. Ideally SSD or NVMe or similar. The faster the better. It does not need to be _very_ large and should not grow in usage _very_ fast, though `stellar-core` will fail if it fills up, so keep an eye on its utilization and make sure there's plenty of room.
+  2. The `DATABASE` config value controls not only which _kind_ of database the node is performing transactions against, but also _where_ the database is located. Unlike with many database-backed programs, the _content_ of the database in a `stellar-core` installation is somewhat ephemeral: every node has a complete copy of it, as does every history archive, and the database can always be restored / rebuilt from history archives (it is in fact being continuously backed up every 5 minutes). So the main thing to optimize for here is latency, especially on nodes doing consensus. We recommend either:
+    * SQLite on a fast, local disk. This is probably the fastest option and is perfectly adequate for many types of node.
```
Comment (tomquisel) [inline docs/software/performance.md]: ditto

**Comment by graydon** [comment]
r+ 8683e14

**Pull request merged**